src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -229,25 +229,27 @@ public class SqlMutationEngine : IMutationEngine
         {
             IOutputType outputType = context.Selection.Field.Type;
             string entityName = string.Empty;
-            ObjectType _underlyingFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(outputType);
+            ObjectType underlyingFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(outputType);
             bool isPointMutation;
-            if (GraphQLUtils.TryExtractGraphQLFieldModelName(_underlyingFieldType.Directives, out string? modelName))
+            if (GraphQLUtils.TryExtractGraphQLFieldModelName(underlyingFieldType.Directives, out string? modelName))
             {
                 isPointMutation = true;
                 entityName = modelName;
             }
             else
             {
-                // Absence of model directive on the mutation indicates that we are dealing with a `mutate many`
+                // Model directive is not added to the output type of 'mutate many' mutations.
+                // Thus, absence of model directive here indicates that we are dealing with a 'mutate many'
                 // mutation like createBooks.
                 isPointMutation = false;
 
-                // For a mutation like createBooks which inserts multiple records into the Book entity, the underlying field type is a paginated response
-                // type like 'BookConnection'. To determine the underlying entity name, we have to look at the type of the `items` field which stores a list of items of
-                // the underlying entity's type - here, Book type.
-                IOutputType entityOutputType = _underlyingFieldType.Fields[MutationBuilder.ARRAY_INPUT_ARGUMENT_NAME].Type;
-                ObjectType _underlyingEntityFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(entityOutputType);
-                if (GraphQLUtils.TryExtractGraphQLFieldModelName(_underlyingEntityFieldType.Directives, out modelName))
+                // For a mutation like createBooks which inserts multiple records into the Book entity,
+                // the underlying field type is a paginated response type like 'BookConnection'.
+                // To determine the underlying entity name, we have to look at the type of the `items` field
+                // which stores a list of items of the underlying entity's type - here, Book type.
+                IOutputType entityOutputType = underlyingFieldType.Fields[MutationBuilder.ARRAY_INPUT_ARGUMENT_NAME].Type;
+                ObjectType underlyingEntityFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(entityOutputType);
+                if (GraphQLUtils.TryExtractGraphQLFieldModelName(underlyingEntityFieldType.Directives, out modelName))
```

Comment:
A few lines up you say
```csharp
// Model directive is not added to the output type of 'mutate many' mutations.
// Thus, absence of model directive here indicates that we are dealing with a 'mutate many'
```
and here you try to extract model name. Is that only possible here because you've retrieved the underlying type from the paginated reponse type?

Reply:
Yes.